File: nexus-backend/src/main/java/tn/arctic/nexus/entities/Report.java
package tn.arctic.nexus.entities;


import jakarta.persistence.*;

@Entity
@Table(name = "report")
public class Report {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "thread_id", nullable = false) // thread_id = publication_id
    private Publication publication;

    @Column(nullable = false)
    private String reason;

    @Column(nullable = false)
    private String status;
}

File: nexus-frontend/src/app/core/entities/community/report.ts
import { User } from '../user/user.model';
import { Publication } from './publication';

export class Report {
  id?: number;
  reason!: string;
  status!: string;
  user!: Partial<User>;
  publication!: Publication;

  constructor(data?: Partial<Report>) {
    Object.assign(this, data);
  }
}
